### PR TITLE
Added MacOS support to ease developing/testing

### PR DESCRIPTION
**Pull request by PieterAelse**
## What does this PR accomplish?

As MacOS support for Flutter is working good, it's easy to use MacOS for development with the benefits of:

- Not having to run a simulator/emulator
- or on a real device
- UI can be developed/tested more easily as your window can be resized.

This solves issue https://github.com/WorldHealthOrganization/app/issues/392 

Required changes to the app for it to work on MacOS are visible in the second commit (6ea1305)

## Did you add any dependencies?

Nope. Did update Dart SDK to 2.7.0 though because latest and extension functions 🎉 

## How did you test the change?

**To run on your MacOS device check the [added Readme file](https://github.com/PieterAelse/app/blob/add_macos_support/client/flutter/macos/README.md) in the `macos` folder.**

Ran the app on my MacBook, works as a charm, see (short) video: 
[MacOS support screenrecording (short).zip](https://github.com/WorldHealthOrganization/app/files/4390880/MacOS.support.screenrecording.short.zip)

Screenshot:
MacOS support screenshot

### CODE REVIEW

File: client/flutter/lib/home_page.dart [hunk 1]
```diff
@@ -7,43 +9,56 @@ import 'package:WHOFlutter/pages/who_myth_busters.dart';
...
-            title: S.of(context).travelAdvice,
-            onPressed: () => Navigator.of(context)
-                .push(MaterialPageRoute(builder: (c) => TravelAdvice()))),
-        PageButton(
-            title: S.of(context).shareTheApp,
+      body: Column(
+        mainAxisSize: MainAxisSize.max,
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
```
Comment (SamMousa) [inline client/flutter/lib/home_page.dart]: Why did you change this?

**Comment by SamMousa** [comment]
I like the idea of this (even though I don't have a Mac).
Please revert your whitespace changes though, they make it really hard to review the code.

File: client/flutter/lib/home_page.dart [hunk 2]
```diff
@@ -7,43 +9,56 @@ import 'package:WHOFlutter/pages/who_myth_busters.dart';
...
+          PageButton(
+            title: localizations.shareTheApp,
             lightColor: true,
-            onPressed: () => Share.share(
-                'Check out the official COVID-19 Guide App https://www.who.int/covid-19-app')),
-        PageButton(
-            title: S.of(context).aboutTheApp,
+            onPressed: () async {
+              final url = 'https://www.who.int/covid-19-app';
+              if (Platform.isMacOS) {
```
Comment (SamMousa) [inline client/flutter/lib/home_page.dart]: I'd prefer to instead catch the exception and then either show a message or open the URL.
That way we can handle it properly for all platforms that don't support sharing, this will also be more robust if we add platforms in the future, or if the `Share.share` adds support for MacOS.

File: client/flutter/lib/main.dart [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+
+import './constants.dart';
+import './home_page.dart';
 import 'generated/l10n.dart';
 
-void main() => runApp(MyApp());
+void main() {
+  /// Since macOS is not yet a supported platform in TargetPlatform we need
+  /// to override the default platform to one that is.
+  debugDefaultTargetPlatformOverride ??= TargetPlatform.iOS;
```
Comment (SamMousa) [inline client/flutter/lib/main.dart]: This really should not go in our main code base.

Also https://github.com/flutter/flutter/pull/51519 has been merged, so we should be seeing it soon in flutter source.

 When that's the case devs can run the latest flutter to run the flutter app on macos, while our CI builds stay on stable (because they don't need to build the macos version anyway)

File: client/flutter/lib/main.dart [hunk 4]
```diff
@@ -27,16 +39,17 @@ class _MyAppState extends State<MyApp> {
       localizationsDelegates: [
         GlobalMaterialLocalizations.delegate,
         GlobalWidgetsLocalizations.delegate,
-        S.delegate
+        S.delegate,
       ],
-      supportedLocales: S.delegate.supportedLocales,
+      supportedLocales: Platform.isMacOS ? [Locale('en', '')] : S.delegate.supportedLocales,
```
Comment (SamMousa) [inline client/flutter/lib/main.dart]: We shouldn't have platform specific code in our main code base if we can avoid it. In this case localization is a known issue and we should fix that instead of adding a workaround for 1 platform.

File: client/flutter/lib/home_page.dart (hunk 2 above)
Comment (SamMousa) [reply client/flutter/lib/home_page.dart]: See #442

**Comment by advayDev1** [comment]
Hey - thank you for this, we'll look at it after v1.

We don't want to add complexity if not absolutely needed.
